src/Components/EventDetailsPage.js

```diff
@@ -82,34 +90,102 @@ class EventDetailsPage extends Component {
                 upvote_usernames: [],
                 upvote_userids: [],
                 comments_content: [],
+                comments_user: [],
                 comments_created: []
             };
             newPosts.push(newPost);
             this.setState({posts: newPosts, messageInput: ""});
         })
     }
 
+    toggleShowComments(threadId) {
+        let newShowComments = {};
+        newShowComments = Object.assign(newShowComments, this.state.showComments);
```

Comment:
Hm... I think this should work because you have booleans as values but i dont think for arrays of complex objects it will do deep copies. Would recommend using the lodash library for any more complex deep copies :)